packages/pds/src/actor-store/index.ts

```diff
@@ -0,0 +1,220 @@
+      max: 30000,
+      fetchMethod: async (did) => {
+        const { keyLocation } = await this.getLocation(did)
+        const privKey = await fs.readFile(keyLocation)
+        return crypto.Secp256k1Keypair.import(privKey)
+      },
+    })
+  }
+
+  private async getLocation(did: string) {
+    const didHash = await crypto.sha256Hex(did)
+    const subdir = path.join(this.resources.dbDirectory, didHash.slice(0, 2))
+    const dbLocation = path.join(subdir, `${did}.sqlite`)
+    const keyLocation = path.join(subdir, `${did}.key`)
+    return { subdir, dbLocation, keyLocation }
+  }
+
+  async keypair(did: string): Promise<Keypair> {
+    const got = await this.keyCache.fetch(did)
+    if (!got) {
+      throw new InvalidRequestError('Keypair not found', 'NotFound')
+    }
+    return got
+  }
+
+  async db(did: string): Promise<ActorDb> {
+    const got = await this.dbCache.fetch(did)
+    if (!got) {
+      throw new InvalidRequestError('Repo not found', 'NotFound')
+    }
+    return got
+  }
+
+  async reader(did: string) {
+    const [db, keypair] = await Promise.all([this.db(did), this.keypair(did)])
+    return createActorReader(did, db, keypair, this.resources)
+  }
+
+  async read<T>(did: string, fn: ActorStoreReadFn<T>) {
+    const reader = await this.reader(did)
+    return fn(reader)
+  }
+
+  async transact<T>(did: string, fn: ActorStoreTransactFn<T>) {
+    const [db, keypair] = await Promise.all([this.db(did), this.keypair(did)])
+    return db.transaction((dbTxn) => {
+      const store = createActorTransactor(did, dbTxn, keypair, this.resources)
+      return fn(store)
+    })
+  }
+
+  async create<T>(
+    did: string,
+    keypair: ExportableKeypair,
+    fn: ActorStoreTransactFn<T>,
+  ) {
+    const { subdir, dbLocation, keyLocation } = await this.getLocation(did)
+    // ensure subdir exists
+    await mkdir(subdir, { recursive: true })
+    const exists = await fileExists(dbLocation)
+    if (exists) {
+      throw new InvalidRequestError('Repo already exists', 'AlreadyExists')
+    }
+    const db: ActorDb = getDb(dbLocation)
+    const migrator = getMigrator(db)
+    const privKey = await keypair.export()
+    await Promise.all([
+      await migrator.migrateToLatestOrThrow(),
+      await fs.writeFile(keyLocation, privKey),
+    ])
+
+    const result = await db.transaction((dbTxn) => {
+      const store = createActorTransactor(did, dbTxn, keypair, this.resources)
+      return fn(store)
+    })
+    this.dbCache.set(did, db)
+    return result
+  }
+
+  async destroy(did: string) {
+    const blobstore = this.resources.blobstore(did)
+    if (blobstore instanceof DiskBlobStore) {
+      await blobstore.deleteAll()
+    } else {
+      const db = await this.db(did)
+      const blobRows = await db.db.selectFrom('blob').select('cid').execute()
+      const cids = blobRows.map((row) => CID.parse(row.cid))
+      await Promise.allSettled(cids.map((cid) => blobstore.delete(cid)))
```

Comment:
I agree that'd be really nice, but I don't think that s3 has a mechanism for deleting all based on a prefix :/

Ended up adding a `deleteMany` method and doing chunks of 500

Reply:
Yeah, I'm with ya— it would involve paginating over all items with that prefix using `ListObjects`.  The thought is just that we don't necessarily need the `blob` table to clear out all blobs being stored for that did.  In some ways it might be nice to just paginate over all the items and remove them from the bucket, no need to worry about any discrepancy between what's tracked in the `blob` table and what's actually stored for the user.

Reply:
The new approach you have in there seems great, this is small peas 👍

Reply:
kk cool 👍 

let's keep that in our back pocket tho 👌